### ISSUE TITLE
surface-density in image wrong for ramses

### ISSUE THREAD

**Issue by mkrets** [pynbody/pynbody#427]
Using Ramses data I noticed a problem with pynbody.plot.image.
Comparing the surface-density from the profile with the one in the image gives different results.

This is the profile: (I checked, this is correct.)
![profile](https://user-images.githubusercontent.com/33623841/33761870-d7e7945e-dc0a-11e7-95c5-53b235cb90e5.png)

![wrong_image](https://user-images.githubusercontent.com/33623841/33761875-dafb8a92-dc0a-11e7-9b1e-23b0c98c4c6d.png)

However, if I make a deepcopy, the image is correct.
![correct_image](https://user-images.githubusercontent.com/33623841/33761878-dce6f3fa-dc0a-11e7-8364-b122fe653a4a.png)

I noticed that for the direct loaded data:
`pynbody.plot.image(h1.g, width="30 kpc", log=True, resolution=500, cmap="plasma", vmin=1e5, vmax=1e9, units='Msol kpc^-2')`

`pynbody.plot.image(h1.g, width="30 kpc", log=True, resolution=500, cmap="plasma", vmin=1e5, vmax=1e9, av_z=True)`

is the same (both give the average along the los).

For the deepcopied image theses are not the same.
In other words `units='Msol kpc^-2'` is not working in the first case.

Attached is an example code:
[script.txt](https://github.com/pynbody/pynbody/files/1542299/script.txt)

**Comment by apontzen** [comment]
Thanks for the report. Could you see if PR #428 (branch fix-projected-denouse) resolves this?
